composeApp/src/commonMain/kotlin/org/ooni/probe/ui/results/ResultsViewModel.kt

```diff
@@ -163,5 +213,15 @@ class ResultsViewModel(
         data class DescriptorFilterChanged(val filterType: ResultFilter.Type<Descriptor>) : Event
 
         data class OriginFilterChanged(val filterType: ResultFilter.Type<TaskOrigin>) : Event
+
+        data class SelectResult(val id: ResultModel.Id) : Event
+
+        data class DeselectResult(val id: ResultModel.Id) : Event
+
+        data object CancelSelection : Event
+
+        data object ToggleSelection : Event
+
+        data object DeleteSelectedResults : Event
```

Comment:
We could reuse the DeleteAllClick, rename it into a DeleteClick, and use according to the selection state.